app/services/decidim/extended_socio_demographic_authorization_handler/autocomplete_service.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+module Decidim
+  module ExtendedSocioDemographicAuthorizationHandler
+    class AutocompleteService
+      def initialize(postal_code)
+        @postal_code = postal_code
+      end
+
+      def self.for(postal_code)
+        new(postal_code).call
+      end
+
+      def call
+        return [] if @postal_code.blank?
+
+        return Rails.cache.read(cache_key) if Rails.cache.exist?(cache_key)
+
+        response = JSON.dump(parsed_response(request))
```

Comment:
Why the answer is dumped in json?

Reply:
because i'mm rendering Json on my controller, but i suppose i could make the transition in the controller